STACpopulator/stac_utils.py

```diff
@@ -62,9 +70,100 @@ def collection2literal(collection):
+        return name
+
+
+# DH: There is a question here whether we want to use pystac.Item or not.
+# pystac.Item takes datetime, start_datetime and end_datetime as optional parameters, and then copies them into
+# properties.
+# If we use pystac.Item, we don't have to put start_datetime and end_datetime into Properties, we can let pystac do
+# that.
+class ItemProperties(BaseModel):
+    start_datetime: Optional[dt.datetime] = None
+    end_datetime: Optional[dt.datetime] = None
+    datetime: Optional[dt.datetime] = None
+
+    @field_validator("datetime", mode="before")
+    def validate_datetime(cls, v: Union[dt.datetime, str], values: Dict[str, Any]) -> dt:
+        if v == "null":
+            if not values["start_datetime"] and not values["end_datetime"]:
+                raise ValueError(
+                    "start_datetime and end_datetime must be specified when datetime is null"
+                )
+
+
+class Geometry(TypedDict):
+    type: str
+    coordinates: List[List[List[float]]]
+
+class Asset(BaseModel):
+    href: AnyHttpUrl
+    media_type: Optional[str] = None
+    title: Optional[str] = None
+    description: Optional[str] = None
+    roles: Optional[List[str]] = None
+
+class Link(BaseModel):
+    """
+    https://github.com/radiantearth/stac-spec/blob/v1.0.0/collection-spec/collection-spec.md#link-object
+    """
+
+    href: str = Field(..., alias="href", min_length=1)
+    rel: str = Field(..., alias="rel", min_length=1)
+    type: Optional[str] = None
+    title: Optional[str] = None
+    # Label extension
+    label: Optional[str] = Field(None, alias="label:assets")
+    model_config = ConfigDict(use_enum_values=True)
+
+    def resolve(self, base_url: str) -> None:
+        """resolve a link to the given base URL"""
+        self.href = urljoin(base_url, self.href)
+
+class PaginationMethods(str, AutoValueEnum):
+    """
+    https://github.com/radiantearth/stac-api-spec/blob/master/api-spec.md#paging-extension
+    """
+
+    GET = auto()
+    POST = auto()
+
+
+class PaginationRelations(str, AutoValueEnum):
+    """
+    https://github.com/radiantearth/stac-api-spec/blob/master/api-spec.md#paging-extension
+    """
+
+    next = auto()
+    previous = auto()
+
+class PaginationLink(Link):
+    """
+    https://github.com/radiantearth/stac-api-spec/blob/master/api-spec.md#paging-extension
+    """
+
+    rel: PaginationRelations
+    method: PaginationMethods
+    body: Optional[Dict[Any, Any]] = None
+    merge: bool = False
+
+Links = RootModel[List[Union[PaginationLink, Link]]]
+
+
+class Item(BaseModel):
+    id: str = Field(..., alias="id", min_length=1)
+    geometry: Optional[Geometry] = None
+    bbox: Optional[List[float]] = None
+    properties: ItemProperties
+    assets: Dict[str, Asset] = None
+    stac_extensions: Optional[List[AnyUrl]] = []
+    collection: Optional[str] = None
+    datetime: Optional[dt.datetime] = None  # Not in the spec, but needed by pystac.Item.
+
 
 
 class CFJsonItem:
```

Comment:
I am note able to quite understand what this class is supposed to do....

Regarding conversion of a Pydantic data model, say `item` to a JSON that can be sent to the server with a POST request, I've found this to work just fine:

`j = json.loads(item.model_dump_json(by_alias=True))`